qnexus/models/__init__.py

```diff
@@ -73,14 +74,19 @@ def df(self) -> pd.DataFrame:
 
 
 class Device(BaseModel):
-    """A device in Nexus, work-in-progress"""
+    """A device in Nexus."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
 
     backend_name: str
     device_name: Optional[str]
     nexus_hosted: bool
-    backend_info: BackendInfo
+    backend_info_dict: dict[str, Any]
 
-    model_config = ConfigDict(arbitrary_types_allowed=True)
+    @property
+    def backend_info(self) -> BackendInfo:
+        """The BackendInfo for the Device."""
+        return BackendInfo.from_dict(self.backend_info_dict)
```

Comment:
We were starting to get errors from pydantic related to generating the Pydantic CoreSchema for the BackendInfo class, so I've reverted to storing the BackendInfo dict as part of the model but then keeping the `backend_info` attribute via a property.

Wasn't able to use a private (underscored) member attribute as this led Pydantic to mark it as a PrivateAttr which led to lots of undesirable behaviour. Having `backend_info_dict` was a compromise